UE4SS/src/UE4SSProgram.cpp

```diff
@@ -1117,67 +1117,127 @@ namespace RC
         }
     }
 
-    template <typename ModType>
-    auto start_mods() -> std::string
+    auto UE4SSProgram::read_mods_json(std::string mod_list_file, std::vector<ModData>& mod_data_vector) -> void
     {
-        ProfilerScope();
-        // Part #1: Start all mods that are enabled in mods.txt.
-        Output::send(STR("Starting mods (from mods.txt load order)...\n"));
-
-        std::filesystem::path mods_directory = UE4SSProgram::get_program().get_mods_directory();
-        std::wstring enabled_mods_file{mods_directory / "mods.txt"};
-        if (!std::filesystem::exists(enabled_mods_file))
+        std::string buffer{};
+        glz::parse_error pe = glz::read_file_json(mod_data_vector, mod_list_file, buffer);
+        if (pe)
         {
-            Output::send(STR("No mods.txt file found...\n"));
+            std::string descriptive_error = glz::format_error(pe, buffer);
+
+            size_t count = descriptive_error.size() + 1;
+            wchar_t* converted_method_name = new wchar_t[count];
+
+            size_t num_of_char_converted = 0;
+            mbstowcs_s(&num_of_char_converted, converted_method_name, count, descriptive_error.data(), count);
+
+            auto converted = File::StringViewType(converted_method_name);
+
+            delete[] converted_method_name;
+                
+            Output::send<LogLevel::Error>(STR("{}\n\nError parsing mods.json file, please fix the file...\n"), converted);
         }
-        else
-        {
-            // 'mods.txt' exists, lets parse it
-            std::wifstream mods_stream{enabled_mods_file};
+    }
 
-            std::wstring current_line;
-            while (std::getline(mods_stream, current_line))
+    auto UE4SSProgram::write_mods_json(std::string mod_list_file, std::vector<ModData>& mod_data_vector) -> void
+    {
+        std::string mod_data_buffer;
+        glz::write<glz::opts{.prettify = true}>(mod_data_vector, mod_data_buffer);
+        glz::error_code ec = glz::buffer_to_file(mod_data_buffer, mod_list_file);
+    }
+    
+    auto UE4SSProgram::convert_legacy_mods_file(StringType legacy_mod_list_file, std::vector<ModData>& mod_data_vector) -> void
+    {
+        Output::send(STR("Converting legacy mods.txt to mods.json...\n"));
+        std::wifstream mods_stream = File::open_file_skip_BOM(legacy_mod_list_file);
+        try {
+            mods_stream = File::open_file_skip_BOM(legacy_mod_list_file);
+        } catch (const std::exception& e) {
+            std::cerr << e.what() << std::endl;
+        }
```

Comment:
This is bad, don't throw here, just let the exception fall through to main like we do with all fatal exceptions.
The way you've currently set this up will cause the error to not actually be logged, it will just be ignored and probably crash as you're still continuing to operate on the stream even though there was an error with it causing the stream to be invalid.